### PR TITLE
Add configuration for traffic director serviceBinding resource

### PR DESCRIPTION
**Pull request by maphad**
Add configuration for traffic director serviceBinding resource

part of https://github.com/hashicorp/terraform-provider-google/issues/14039

If this PR is for Terraform, I acknowledge that I have:

- [ x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_network_services_service_binding`
```

### CODE REVIEW

File: mmv1/products/networkservices/ServiceBinding.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    description: Set of label tags associated with the ServiceBinding resource.
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      A free-text description of the resource. Max length 1024 characters.
+  - !ruby/object:Api::Type::String
+    name: 'service'
+    required: true
+    description: |
+      Required. The full Service Directory Service name of the format
```
Comment (slevenick) [inline mmv1/products/networkservices/ServiceBinding.yaml:87]:
```suggestion
      The full Service Directory Service name of the format 
```

File: mmv1/products/networkservices/ServiceBinding.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'ServiceBinding'
+base_url: 'projects/{{project}}/locations/global/serviceBindings'
+create_url: 'projects/{{project}}/locations/global/serviceBindings?serviceBindingId={{name}}'
+self_link: 'projects/{{project}}/locations/global/serviceBindings/{{name}}'
+min_version: beta
+update_verb: :PATCH
+update_mask: true
```
Comment (slevenick) [inline mmv1/products/networkservices/ServiceBinding.yaml:21]: Can you add a handwritten update test to make sure this works? There are examples of tests in https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests

File: mmv1/products/networkservices/ServiceBinding.yaml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  ServiceBinding is the resource that defines a Service Directory Service to be used in a 
+  BackendService resource.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  api: 'https://cloud.google.com/traffic-director/docs/reference/network-services/rest/v1beta1/projects.locations.serviceBindings'
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
```
Comment (slevenick) [inline mmv1/products/networkservices/ServiceBinding.yaml:32]: These seem very long for such a simple resource. Does this actually take a long time to provision?

**Review by slevenick — CHANGES_REQUESTED** [review]

File: mmv1/products/networkservices/ServiceBinding.yaml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'ServiceBinding'
+base_url: 'projects/{{project}}/locations/global/serviceBindings'
+create_url: 'projects/{{project}}/locations/global/serviceBindings?serviceBindingId={{name}}'
```
[the remark this answers is not in the document]
Comment (maphad) [reply mmv1/products/networkservices/ServiceBinding.yaml:17]: It is going to stay a global resource in the long term.

File: mmv1/products/networkservices/ServiceBinding.yaml (hunk 3 above)
Comment (maphad) [reply mmv1/products/networkservices/ServiceBinding.yaml:29]: You are right, I reduced the timeouts now.

File: mmv1/products/networkservices/ServiceBinding.yaml (hunk 2 above)
Comment (maphad) [reply mmv1/products/networkservices/ServiceBinding.yaml]: Thanks for catching this. Update on this resource is not supported. I fixed the yaml now.

**Comment by maphad** [comment]
> Can you add a handwritten update test to make sure we can update the updatable fields correctly? If a field is not updatable then add immutable: true to it

The "update" API is not supported for this resource. Hence, I skipped the update test.

**Comment by slevenick** [comment]
> > Can you add a handwritten update test to make sure we can update the updatable fields correctly? If a field is not updatable then add immutable: true to it
> 
> The "update" API is not supported for this resource. Hence, I skipped the update test.

Then can you add immutable: true for the resource at the top-level? It will make any change force a recreation of the resource

**Comment by maphad** [comment]
> > > Can you add a handwritten update test to make sure we can update the updatable fields correctly? If a field is not updatable then add immutable: true to it
> > 
> > 
> > The "update" API is not supported for this resource. Hence, I skipped the update test.
> 
> Then can you add immutable: true for the resource at the top-level? It will make any change force a recreation of the resource

Ah makes sense! Added...thanks for this suggestion!

**Review by slevenick — APPROVED** [review]

**Pull request merged**